### PR TITLE
REF: useExtendedNavigation

### PR DESCRIPTION
**Pull request by marcosrdz**
The current way we handle biometric unlock per screen and other prerequisites (such as push auth) is breaking the native stack (especially on macOS). This refactoring checks which screen is requested, checks if it requires bio..asks for it, then proceeds with the push

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2ks7t6sxkbevhtvwcnvfd24ceu?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/911)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/w4ngqwjkd62ujms37vb6x2daiy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/915)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/37mtgxtltnd74x4szcfmk7cyni?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/916)

File: components/NavigationProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React, { ReactNode } from 'react';
```
Comment (Overtorment) [inline components/NavigationProvider.tsx:1]: all this navigation fuckery is a mystery tome

**Review by Overtorment — APPROVED** [review]
lgtm

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ubmvnz7kvfc6hqcsrej6oqbkd4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/918)

File: components/NavigationProvider.tsx (hunk 1 above)
Comment (marcosrdz) [reply components/NavigationProvider.tsx:1]: I wanted to provide a hook that will auto handle requirement logic instead of having to manually deal with it

**Review by limpbrains — APPROVED** [review]
Nice approach

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**